src/main/java/com/traveltime/sdk/TravelTimeSDK.java

```diff
@@ -109,14 +113,56 @@
         return future;
     }
 
-
     public <T> Either<TravelTimeError, T> sendProto(ProtoRequest<T> request) {
         return request
             .createRequest(baseProtoUri, credentials)
             .flatMap(this::executeRequest)
             .flatMap(response -> deserializeResponse(request, response));
     }
 
+    private static <T> Either<TravelTimeError, T> getFuture(CompletableFuture<Either<TravelTimeError, T>> future) {
+        try {
+            return future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            return Either.left(new IOError(e, "Error while executing an async request."));
+        }
+    }
+
+    public Either<TravelTimeError, TimeFilterFastProtoResponse> sendProtoBatched(
+        TimeFilterFastProtoRequest request,
+        int batchSizeHint
+    ) {
+        return getFuture(sendProtoAsyncBatched(request, batchSizeHint));
+    }
+
+    public CompletableFuture<Either<TravelTimeError, TimeFilterFastProtoResponse>> sendProtoAsyncBatched(
+        TimeFilterFastProtoRequest request,
+        int batchSizeHint
```

Comment:
Should we have a default batch size of, say, 100k?

Reply:
I think we should default to 100k. We have benchmarked with 100k in mind.

Reply:
We've benchmarked with 100k as the ceiling, the optimal default batch size might be something different.